crates/uv/src/commands/python/install.rs

```diff
@@ -354,32 +354,53 @@ pub(crate) async fn install(
+                        Ok(valid) => valid,
+                        Err(err) => {
+                            debug!("Failed to inspect executable with error: {err}");
+                            true // Treat this as a valid link, we won't remove it without `--force`
+                        }
+                    };
+
+                    // Figure out what installation it references, if anyf
+                    let existing = valid_link
+                        .then(|| {
+                            find_matching_bin_link(
+                                installations
+                                    .iter()
+                                    .copied()
+                                    .chain(existing_installations.iter()),
+                                &target,
+                            )
+                        })
+                        .flatten();
 
                     match existing {
                         None => {
                             // There's an existing executable we don't manage, require `--force`
-                            if !force {
-                                errors.push((
-                                    installation.key(),
-                                    anyhow::anyhow!(
-                                        "Executable already exists at `{}` but is not managed by uv; use `--force` to replace it",
-                                        to.simplified_display()
-                                    ),
-                                ));
-                                continue;
+                            if valid_link {
+                                if !force {
+                                    errors.push((
+                                        installation.key(),
+                                        anyhow::anyhow!(
+                                            "Executable already exists at `{}` but is not managed by uv; use `--force` to replace it",
+                                            to.simplified_display()
+                                        ),
+                                    ));
+                                    continue;
+                                }
+                                debug!(
+                                    "Replacing existing executable at `{}` due to `--force`",
+                                    target.simplified_display()
+                                );
+                            } else {
+                                debug!(
+                                    "Replacing existing executable at `{}` due to an invalid symlink",
+                                    target.simplified_display()
+                                );
                             }
-                            debug!(
-                                "Replacing existing executable at `{}` due to `--force`",
-                                target.simplified_display()
-                            );
```

Comment:
This logic only runs on Linux, right? It doesn't look like it's gated, but I assume the error it's handling only runs on Linux?

Reply:
I think this technically would run on Windows too (which is valid, you _could_ have a broken symlink it's just rare).